test/e2e/framework/framework.go

```diff
@@ -119,6 +134,12 @@ func (f *Framework) setup() error {
 	}
 	logrus.Info("etcd backup operator created successfully")
 
+	err = f.SetupEtcdRestoreOperatorAndService()
+	if err != nil {
+		return fmt.Errorf("failed to create etcd restore operator pod and service: %v", err)
```

Comment:
This error is too confusing.
I would suggest just `return err` here and make the explicit error inside the func where it failed